src/swupd_lib/target_root.c

```diff
@@ -330,6 +333,61 @@ static enum swupd_code install_dir(const char *fullfile_path, const char *target
 	return install_dir_using_tar(fullfile_path, target_file);
 }
 
+/* Backup target directory to a backup and timestamp prefixed name. */
+/* This function should only be used after the deletes are processed. */
+static enum swupd_code move_to_timestamp(const char *filename)
+{
+	struct tm *timeinfo;
+	time_t now;
+	char time_str[50] = { 0 };
+	char *backup_dir = NULL;
+	int ret = SWUPD_OK;
+	char *target_basename;
+	char *dir = NULL;
+	char *target_file = NULL;
+	char *target_path = NULL;
+
+	dir = sys_dirname(filename);
+	target_basename = sys_basename(filename);
+	target_path = sys_path_join("%s/%s", globals.path_prefix, dir);
+	target_file = sys_path_join("%s/%s", globals.path_prefix, filename);
+
+	time(&now);
+	timeinfo = localtime(&now);
+	if (!timeinfo) {
+		ret = SWUPD_TIME_UNKNOWN;
+		goto out;
+	}
+
+	strftime(time_str, sizeof(time_str), "%s", timeinfo);
+	backup_dir = sys_path_join("%s/%s%s.%s", target_path, BACKUP_FILE_PREFIX, time_str, target_basename);
+
+	/* It shouldn't be possible to have an already created backup_dir
+	   but just in case, try and remove it and error out if unable to. */
+	ret = sys_rm_recursive(backup_dir);
```

Comment:
shuold we actually check if this is there first? just incase we have some bug... then it only ever happens if it actually exists

Reply:
Sure, less to go wrong. I suspect if things are going wrong in this area though we are in a whole different world of hurt that'll show up all over in testing.